controllers/akodeploymentconfig/akodeploymentconfig_controller.go

```diff
@@ -48,6 +49,7 @@ type AKODeploymentConfigReconciler struct {
 	Scheme            *runtime.Scheme
 	userReconciler    *user.AkoUserReconciler
 	ClusterReconciler *cluster.ClusterReconciler
+	network_provider.UsableNetworkProvider
```

Comment:
Better not use the name like this, golang has his own package name convention https://go.dev/blog/package-names

Reply:
any idea of more ideal name `netprovider` for example?

Reply:
`networkprovider` or `netprovider` is fine